sdk/core/abort-controller/package.json

```diff
@@ -10,7 +10,7 @@
     "build:samples": "echo Obsolete",
     "build:test": "tsc -p . && rollup -c 2>&1",
     "build:types": "downlevel-dts types/src types/3.1",
-    "build": "npm run clean && tsc -p . && rollup -c 2>&1 && npm run build:types",
+    "build": "npm run clean && npm run extract-api && rollup -c 2>&1 && npm run build:types",
```

Comment:
We still need `tsc -p .` I believe.

The `files` section can now be updated to have the bundled `.d.ts` files

Reply:
@jeremymeng the `extract-api` script already runs `tsc`. 

I agree on updating the files list. It includes all files under `types` in addition to a shim file, so I would suggest to keep the shim and remove `types` entries. Furthermore, export everything from the d.ts file created by the api extractor from the shim (see [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/schemaregistry/schema-registry-avro/types/schema-registry-avro.shims.d.ts) for example). Actually since downlevel types is being used, you will need to have two such shims, one for < v3.6 and another for everything else (see [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/core-rest-pipeline.shims-3_1.d.ts) for example).

Reply:
> @jeremymeng the `extract-api` script already runs `tsc`.

Ah that's right! This is running the npm script, not the api-extractor command.  On a side note, our build scripts are not consistent in this.  I will see if we have an issue tracking that.

Reply:
I did some work on this for core packages in https://github.com/Azure/azure-sdk-for-js/pull/16248. This change does not align with the standardization I did. @ramya-rao-a could you consider aligning?

Reply:
The template package has the below

> npm run clean && tsc -p . && rollup -c 2>&1 && api-extractor run --local && npm run build:types

Is this what you are referring to @deyaaeldeen ?

Reply:
@ramya-rao-a yes!

Reply:
Done

Reply:
@deyaaeldeen, @jeremymeng Can either of you create an issue to track the files ask?